pkg/controller/groups/variables/controller.go

```diff
@@ -0,0 +1,233 @@
+		return nil, err
+	}
+	return &external{kube: c.kube, client: c.newGitlabClientFn(*cfg)}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client groups.VariableClient
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.Variable)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotVariable)
+	}
+	if cr.Spec.ForProvider.GroupID == nil {
+		return managed.ExternalObservation{}, errors.New(errGroupIDMissing)
+	}
+
+	variable, res, err := e.client.GetVariable(
+		*cr.Spec.ForProvider.GroupID,
+		cr.Spec.ForProvider.Key,
+		gitlab.WithContext(ctx))
+
+	if err != nil {
+		if clients.IsResponseNotFound(res) {
+			return managed.ExternalObservation{}, nil
+		}
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetFailed)
+	}
+
+	if cr.Spec.ForProvider.ValueSecretRef != nil {
+		if err = e.updateVariableFromSecret(ctx, cr.Spec.ForProvider.ValueSecretRef, &cr.Spec.ForProvider); err != nil {
+			cr.Status.SetConditions(xpv1.ReconcileError(err))
+			return managed.ExternalObservation{}, errors.Wrap(err, errUpdateFailed)
+		}
+	}
+
+	current := cr.Spec.ForProvider.DeepCopy()
+	groups.LateInitializeVariable(&cr.Spec.ForProvider, variable)
+
+	cr.Status.SetConditions(xpv1.Available())
+
+	return managed.ExternalObservation{
+		ResourceExists:          true,
+		ResourceUpToDate:        groups.IsVariableUpToDate(&cr.Spec.ForProvider, variable),
+		ResourceLateInitialized: !cmp.Equal(current, &cr.Spec.ForProvider),
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.Variable)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotVariable)
+	}
+
+	if cr.Spec.ForProvider.ValueSecretRef != nil {
+		if err := e.updateVariableFromSecret(ctx, cr.Spec.ForProvider.ValueSecretRef, &cr.Spec.ForProvider); err != nil {
+			cr.Status.SetConditions(xpv1.ReconcileError(err))
+			return managed.ExternalCreation{}, errors.Wrap(err, errCreateFailed)
+		}
+	}
+	if cr.Spec.ForProvider.GroupID == nil {
+		return managed.ExternalCreation{}, errors.New(errGroupIDMissing)
+	}
+
+	cr.Status.SetConditions(xpv1.Creating())
+	_, _, err := e.client.CreateVariable(
+		*cr.Spec.ForProvider.GroupID,
+		groups.GenerateCreateVariableOptions(&cr.Spec.ForProvider),
+		gitlab.WithContext(ctx))
+
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateFailed)
+	}
+	return managed.ExternalCreation{}, nil
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.Variable)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotVariable)
+	}
+
+	if cr.Spec.ForProvider.ValueSecretRef != nil {
+		if err := e.updateVariableFromSecret(ctx, cr.Spec.ForProvider.ValueSecretRef, &cr.Spec.ForProvider); err != nil {
+			cr.Status.SetConditions(xpv1.ReconcileError(err))
```

Comment:
This is not necessary. The ReconcileError status condition is set automatically when you return an error during `Oberve`, `Create`, `Update` and `Delete`.

```suggestion
```